src/classes/VOL_CTRL_PersonalSiteContactInfo.cls

```diff
@@ -258,198 +269,223 @@ global with sharing class VOL_CTRL_PersonalSiteContactInfo {
+                    dtStart = hr.Start_Date__c;
+                    hr.Comments__c = dtStart.formatGmt(strDateFormat);
+                } else {
+                    double duration = hr.Hours_Worked__c == null ? hr.Volunteer_Shift__r.Duration__c : hr.Hours_Worked__c;
+                    DateTime dtEnd = dtStart.addMinutes(integer.valueOf(duration * 60));
+                    string strStart = dtStart.format(strFormat, strTimeZone);
+                    
+                    // see if start and end are on the same day
+                    if (dtStart.format('d', strTimeZone) == dtEnd.format('d', strTimeZone)) {
+                        hr.System_Note__c =  dtStart.format(strFormatEndTime, strTimeZone) + ' - ' + dtEnd.format(strFormatEndTime, strTimeZone);   
+                    } else {
+                        hr.System_Note__c =  dtStart.format(strFormatEndTime, strTimeZone) + ' - ' + dtEnd.format(strFormat, strTimeZone);                      
+                    } 
+                    
+                    // also save user formated Start Date in Comments field
+                    hr.Comments__c = dtStart.format(strDateFormat, strTimeZone);
+                }           
+            }
+        } catch (Exception e) {
+            ApexPages.addMessages(e);                 
+        }
     }
         
-	global PageReference edit() {
-		isEditing = true;    
-		return null;
-	}
+    global PageReference edit() {
+        isEditing = true;    
+        return null;
+    }
 
-	global PageReference save() {  
-        if (contactReadOnly != null) {        	
-        	//Contact con = listCon[0];       	
-            //  now copy over all the non-null fields from the form's contact to the existing contact.
-            for (string strF : listStrContactFields) {
-            	if (contactEdit.get(strF) != null) {
-            		try {
-            			contactReadOnly.put(strF, contactEdit.get(strF));
-            		} catch(exception ex) {
-            			
-            		}
-            	}
-            }            
-            // save the changes
-            update contactReadOnly;
+    global PageReference save() { 
+        try { 
+            if (contactReadOnly != null) {          
+                //Contact con = listCon[0];         
+                //  now copy over all the non-null fields from the form's contact to the existing contact.
+                for (string strF : listStrContactFields) {
+                    if (contactEdit.get(strF) != null) {
+                        try {
+                            contactReadOnly.put(strF, contactEdit.get(strF));
+                        } catch(exception ex) {
+                            
+                        }
+                    }
+                }
+                // Ensure the user can update the object's fields 
+                UTIL_Describe.checkUpdateAccess('Contact', new Set<String>{'Volunteer_Status__c'});            
```

Comment:
@kbromer @njjc @joy-sfdo As i feared, adding the permissions checks to the PersonalSiteContactInfo page has made it so the Sites Guest User can no longer update the Contact object.  Even though the permissions are enabled for this profile, Salesforce started preventing it in API 26.  There is a hack in V4S that keeps one module at API 25 just to work around this issue.  But with the inclusion of the new FLS checks Joy has put into this class, we have run into it again.  I'd like to propose we just remove the Access checks to Contact in this class, and see if we can skate by security review.  The only other choice is to move the save logic into another module and hope they don't notice its using an older API.  i'm also not 100% positive that will work.